### PR TITLE
[WIP] Cancel Bounty/ Why Contributor Stopped Work

### PR DESCRIPTION
**Pull request by octavioamu**
##### Checklist

- [ ] Read and conforms to the [contributor guidelines](https://docs.gitcoin.co/mk_contributors/)
- [ ] Read and conforms to the [style guidelines](https://docs.gitcoin.co/mk_styleguide/)
- [ ] Linter status: 100% pass
- [ ] Added relevant tests
- [ ] Tests pass and code coverage has not decreased
- [ ] Changes don't break existing behavior
- [ ] Commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)
- [ ] [Code documentation](https://docs.gitcoin.co/mk_contributors/#docstrings)
- [ ] Completed: Code review by core team and any requested changes

##### Refers/Fixes

Refs: #2668
###### Contributor

- [ ] Read and followed the [Contributor Guidelines](https://docs.gitcoin.co/mk_contributors/)
- [ ] Tested all changes **locally**
- [ ] Verified existing functionality
- [ ] Ran `make test` and everything passed!

###### Reviewer

- [ ] Affirm contributor guidelines have been followed and requested changes made
- [ ] CI tests and linting pass
- [ ] No conflicts (migrations, files, etc)
- [ ] Regression tested against staging

###### Funder

- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

### CODE REVIEW

**Comment by octavioamu** [comment]
Great changes @mbeacom Thanks for that.

**Comment by thelostone-mc** [comment]
@octavioamu  could you address the stickler comments ? 
Other than that LGTM. will pull it in and check post the update

File: app/dashboard/views.py [hunk 1]
```diff
@@ -264,7 +264,7 @@ def new_interest(request, bounty_id):
             status=401)
 
     try:
-        Interest.objects.get(profile_id=profile_id, bounty=bounty)
+        Interest.objects.exclude(status__in=Interest.INACTIVE_STATUSES).get(profile_id=profile_id, bounty=bounty)
```
Comment (octavioamu) [inline app/dashboard/views.py]: @SaptakS can you help me with  this one? I think create_new_interest_helper need to be updated so if the user stop work and the start again this update the entry and not create a new one. Does that make sense?

Comment (thelostone-mc) [reply app/dashboard/views.py]: @octavioamu can't we use `active()` here ?

File: app/dashboard/views.py [hunk 2]
```diff
@@ -415,25 +422,26 @@ def extend_expiration(request, bounty_id):
     }, status=200)
 
 
-@csrf_exempt
+# @csrf_exempt
```
Comment (thelostone-mc) [inline app/dashboard/views.py]: @SaptakS why was this disabled in the first place?  any idea ?
@octavioamu could we remove it ?

File: app/dashboard/views.py [hunk 3]
```diff
@@ -415,25 +422,26 @@ def extend_expiration(request, bounty_id):
...
+        canceled_bounty_reason (string): The reason for the cancellation.
 
     Returns:
         dict: The success key with a boolean value and accompanying error.
 
     """
-    print(request.POST.get('canceled_bounty_reason'))
     user = request.user if request.user.is_authenticated else None
+    reason = request.POST.get('canceled_bounty_reason', '')
+    print(request.POST.get('canceled_bounty_reason'))
```
Comment (thelostone-mc) [inline app/dashboard/views.py]: remove this ?

File: app/dashboard/views.py (hunk 1 above)
Comment (octavioamu) [reply app/dashboard/views.py]: @SaptakS I resolved this by mistake but I need your helpo about that BE logic.

**Comment by PixelantDesign** [comment]
**Engineering needs to decide**
- Need to decide how this affects business rules. 
- Workers can start work more than once.
- How to express interest modal works when stopped work is removed
- Needs testing
- Resolve Conflicts
- Deploy

cc @danlipert @thelostone-mc

**Comment by thelostone-mc** [comment]
Closing this out ! 
Keeping the branch alive though. Will reopen if needed

**Pull request closed without merging**